src/deploy-cromwell-on-azure/KubernetesManager.cs

```diff
@@ -0,0 +1,261 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using k8s;
+using Microsoft.Azure.Management.ContainerService;
+using Microsoft.Azure.Management.ContainerService.Fluent;
+using Microsoft.Azure.Management.Msi.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Authentication;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+using Microsoft.Azure.Management.Storage.Fluent;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CromwellOnAzureDeployer
+{
+    /// <summary>
+    ///     Class to hold all the kubernetes specific deployer logic. 
+    /// </summary>
+    internal class KubernetesManager
+    {
+        private readonly string BlobCsiRepo = "https://raw.githubusercontent.com/kubernetes-sigs/blob-csi-driver/master/charts";
+        private readonly string BlobCsiDriverVersion = "v1.15.0";
+        private readonly string AadPluginRepo = "https://raw.githubusercontent.com/Azure/aad-pod-identity/master/charts";
+        private readonly string AadPluginVersion = "4.1.12";
+
+        private Configuration configuration { get; set; }
+        private AzureCredentials azureCredentials { get; set; }
+        private CancellationTokenSource cts { get; set; }
+
+        public KubernetesManager(Configuration config, AzureCredentials credentials, CancellationTokenSource cts)
+        {
+            this.configuration = config;
+            this.azureCredentials = credentials;
+            this.cts = cts;
+        }
+
+        public async Task<IKubernetes> GetKubernetesClient(IResource resourceGroupObject)
+        {
+            var resourceGroup = resourceGroupObject.Name;
+            var containerServiceClient = new ContainerServiceClient(azureCredentials) { SubscriptionId = configuration.SubscriptionId };
+
+            // Write kubeconfig in the working directory, because KubernetesClientConfiguration needs to read from a file, TODO figure out how to pass this directly. 
+            var creds = await containerServiceClient.ManagedClusters.ListClusterAdminCredentialsAsync(resourceGroup, configuration.AksClusterName);
+            var kubeConfigFile = new FileInfo("kubeconfig.txt");
+            var contents = Encoding.Default.GetString(creds.Kubeconfigs.First().Value);
+            var writer = kubeConfigFile.CreateText();
+            writer.Write(contents);
+            writer.Close();
+
+            var k8sConfiguration = KubernetesClientConfiguration.LoadKubeConfig(kubeConfigFile, false);
+            var k8sConfig = KubernetesClientConfiguration.BuildConfigFromConfigObject(k8sConfiguration);
```

Comment:
(Minor) Suggesting name change for k8sConfig to kubernetesClientConfiguration or k8sClientConfiguration, for clarity.

Reply:
Fixed